src/pulselistener/pulselistener/mercurial.py

```diff
@@ -192,13 +189,10 @@ class MercurialWorker(object):
     '''
     Mercurial worker maintaining a local clone of mozilla-unified
     '''
-    def __init__(self, phabricator_api, publish_phabricator, cache_root, repositories):
+    def __init__(self, queue_name, queue_phabricator, phabricator_api, cache_root, repositories):
```

Comment:
What if we make Mercurial totally independent from Phabricator? (with the "main" of the service handling the communication between the two)

Main <-> WebServer
Main <-> Mercurial
Main <-> Phabricator

This way different services can use Mercurial in different ways

Reply:
As discussed on IRC, that's the next PR :smile: